src/ddsrt/include/dds/ddsrt/cdtors.h

```diff
@@ -14,16 +14,40 @@
 #include "dds/export.h"
 #include "dds/ddsrt/sync.h"
 
+/**
+ * @file cdtors.h
+ * @brief Contains initializing and finalizing functions for some ddsrt units, and provides a singleton mutex and cond.
+ * 
+ * Initializes and finalizes mutexes of ddsrt/random and ddsrt/atomics, and for Windows also the ddsrt/socket
+ * (not necessary for other platforms).
+ */
+
 #if defined (__cplusplus)
 extern "C" {
 #endif
 
+/**
+ * @brief Initialize ddsrt units
+ */
 void ddsrt_init(void);
 
+/**
+ * @brief Finalize ddsrt units
```

Comment:
This then also needs a mention of the reference counting and note that only the last one actually de-initializes it.

Moreover: it does not allow for calling `init` in parallel to the `fini` call that de-initializes everything. Arguably, that's a bug, but not a urgent one.